pkg/execution/state/state.go

```diff
@@ -305,7 +305,7 @@ type Mutater interface {
 	UpdateMetadata(ctx context.Context, runID ulid.ULID, md MetadataUpdate) error
 
 	// Delete removes state from the state store.
-	Delete(ctx context.Context, i Identifier) error
+	Delete(ctx context.Context, i Identifier) (bool, error)
```

Comment:
cc @BrunoScheufler - sharing context as it's a change to the state interfaces we'll need to account for when we upstream this.